base/logging.go

```diff
@@ -483,6 +483,25 @@ func UpdateLogger(logFilePath string) {
 	}
 }
 
+// This provides an io.Writer interface around the base.Log API
+type LoggerWriter struct {
+	LogKey string  // The log key to log to, eg, "HTTP++"
+
+}
+
+// Write() method to satisfy the io.Writer interface
+func (lw *LoggerWriter) Write(p []byte) (n int, err error) {
+	printf(fgYellow+lw.LogKey+": "+reset+"%s", string(p))
+	return len(p), nil
```

Comment:
Great catch!  I changed it to pass in the required params and emit them in the logs